### ISSUE TITLE
Variable subelement tag

### ISSUE THREAD

**Issue by esbullington** [zpnester/bs-xml#1]
Do you have any suggestions on how to decode an XML document with variable subelements. For example, a particular parent tag may contain:

    <parent-tag>
        <subelement-one/>
    </parent-tag>
or

    <parent-tag>
        <subelement-two/>
    </parent-tag>
or

    <parent-tag>
        <subelement-three/>
    </parent-tag>

I've been trying variations of `Xml.Decode.oneOf` but I can't get anything to work (I'm trying to parse them into something analogous to a variant of type ```type subElements = | SubElementOne | SubElementTwo | SubElementThree;``` ). 

```
let parseSubelements = elem =>
    elem
    |> oneOf([
         elem => {
           elem->withName("subelement-one") |> ignore;
           SubElementOne;
         },
         elem => {
           elem->withName("subelement-two") |> ignore;
           SubElementTwo;
         },
         elem => {
           elem->withName("subelement-three") |> ignore;
           SubElementThree;
          },
    ]);
```

It compiles, but doesn't work.  It only produces an `undefined` instead of the parsed subelement in question.

Essentially, I need a way to check what the child tag name of a given element is. I can get the cdata of an element with `text`, but the only way I see to acquire the child element requires me to pass in the child element name, eg `child("subelement-one", text)` which excludes a varying child. With `name` I get the parent element, but not the variable child. Any suggestions? 

Thanks!

Edit: Looking at the source, I think I'd need another function to get the name of the current element's child(ren), unless there's some feature I'm missing. Would you accept a PR? Maybe called `firstChildName` and/or `childNames`?

**Comment by zpnester** [comment]
Such feature is missing indeed. I'll try to publish an update tomorrow.

**Comment by zpnester** [comment]
Added Xml.Decode.childElements function (version 0.2.4), it should help.

Currently API is a little bit messy, it requires some time to improve it.
No PR needed, I will look into it.

**Comment by zpnester** [comment]
Published 0.3.0.

Unfortunately this update contains breaking changes. 
[Changelog](https://github.com/zpnester/bs-xml/blob/master/CHANGELOG.MD)

**Comment by esbullington** [comment]
Great, thanks for that.
